### PR TITLE
Examples Updates

### PR DESCRIPTION
**Pull request by py-rh**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,11 +46,11 @@ def sd_generate(prompt, **inference_kwargs):
     return model(prompt, **inference_kwargs).images
 
 if __name__ == "__main__":
-    gpu = rh.cluster(name="rh-a10x", instance_type="A10G:1", provider="aws").up_if_not()
-    sd_env = rh.env(reqs=["torch", "transformers", "diffusers"], name="sd_env")
+    img = rh.Image("sd_image").install_packages(["torch", "transformers", "diffusers"])
+    gpu = rh.cluster(name="rh-a10x", instance_type="A10G:1", provider="aws", image = img).up_if_not()
```
Comment (carolineechen) [inline README.md:50]:
```suggestion
    gpu = rh.cluster(name="rh-a10x", instance_type="A10G:1", provider="aws", image=img).up_if_not()
```

File: examples/dask-preprocessing-and-training/dask_on_ray.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            "dask-ml",
+            "dask[distributed]",
+            "dask[dataframe]",
+            "boto3",
+            "s3fs",
+            "xgboost",
+        ]
+    )
+
+    cluster_name = f"py-{num_nodes}"
```
Comment (carolineechen) [inline examples/dask-preprocessing-and-training/dask_on_ray.py:47]: nit and idrc but does the py stand for your initials / do we wanna rename? in a couple of places

File: examples/dask-preprocessing-and-training/permissions.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+title: Minimal Skypilot Permissions
```
Comment (carolineechen) [inline examples/dask-preprocessing-and-training/permissions.yaml:1]: what is this used for?

File: examples/hpo/hpo.py [hunk 4]
```diff
@@ -22,8 +22,8 @@ async def find_best_params():
     cluster = rh.cluster(
         name="rh-4x16-cpu", instance_type="CPU:16", num_nodes=4, provider="aws"
     ).up_if_not()
-    train_env = rh.env(name="worker_env", compute={"CPU": 8})
-    remote_train_fn = rh.function(train_fn).to(cluster, env=train_env)
+
```
Comment (carolineechen) [inline examples/hpo/hpo.py:25]: is the compute important to keep or nah

File: examples/lightning-resnet/resnet_training.py [hunk 5]
```diff
@@ -0,0 +1,274 @@
+# from runhouse.resources.images.image import Image
```
Comment (carolineechen) [inline examples/lightning-resnet/resnet_training.py:1]: can remove this comment

File: examples/parallel-hf-embedding-ec2/parallel_hf_embedding_ec2.py [hunk 6]
```diff
@@ -149,46 +149,50 @@ def embed_docs(self, url: str, **embed_kwargs):
...
-        ),
+        image=img,
     ).up_if_not()
 
-    # Generally, when using Runhouse, you would initialize an env with `rh.env`, and send your module to
-    # that env. Each env runs in a *separate process* on the cluster. In this case, we want to have 4 copies of the
-    # embedding model in separate processes, because we have 4 GPUs. We can do this by creating 4 separate envs
-    # and 4 separate modules, each sent to a separate env. We do this in a loop here, with a list of dependencies
+    # Generally, when using Runhouse, you would initialize an image with `rh.Image`, and send your module to
+    # processes. Each process runs in a *separate process* on the cluster. In this case, we want to have 4 copies of the
```
Comment (carolineechen) [inline examples/parallel-hf-embedding-ec2/parallel_hf_embedding_ec2.py:173]:
```suggestion
    # a process. Each process runs in a *separate process* on the cluster. In this case, we want to have 4 copies of the
```

File: examples/parallel-hf-embedding-ec2/parallel_hf_embedding_ec2.py [hunk 7]
```diff
@@ -149,46 +149,50 @@ def embed_docs(self, url: str, **embed_kwargs):
...
+    # embedding model in separate processes, because we have 4 GPUs. We can do this by creating 4 separate processes
+    # and 4 separate modules, each sent to a separate processes. We do this in a loop here, with a list of dependencies
```
Comment (carolineechen) [inline examples/parallel-hf-embedding-ec2/parallel_hf_embedding_ec2.py:175]:
```suggestion
    # and 4 separate modules, each sent to a separate process. We do this in a loop here, with a list of dependencies
```

File: examples/parallel-hf-embedding-ec2/parallel_hf_embedding_ec2.py [hunk 8]
```diff
@@ -149,46 +149,50 @@ def embed_docs(self, url: str, **embed_kwargs):
...
     # that we need remotely to run the module.
     #
-    # In this case, each `env` is also on a separate machine, but we could also provision an A10G:4 instance,
-    # and send all 4 envs to the same machine. Each env runs within a separate process on the machine, so they
+    # In this case, each `processes` is also on a separate machine, but we could also provision an A10G:4 instance,
```
Comment (carolineechen) [inline examples/parallel-hf-embedding-ec2/parallel_hf_embedding_ec2.py:178]:
```suggestion
    # In this case, each `process` is also on a separate machine, but we could also provision an A10G:4 instance,
```

File: examples/pytorch-resnet/resnet_training.py [hunk 9]
```diff
@@ -262,6 +280,7 @@ def predict(self, image):
         provider="aws",
         image=img,
     ).up_if_not()
+    gpu_cluster.restart_server()
```
Comment (carolineechen) [inline examples/pytorch-resnet/resnet_training.py:283]: remove this?

File: examples/pytorch-torchvision-mnist-training/airflow-multicloud/local_run_of_callables.py [hunk 10]
```diff
@@ -26,18 +26,17 @@
 
     logger.info("Step 1: Bring up cluster")
     cpu_cluster_config = {
-        "cluster_name": "cpu-cluster",
-        "instance_type": "CPU:4+",
+        "cluster_name": "cpu-cluster-test",
```
Comment (carolineechen) [inline examples/pytorch-torchvision-mnist-training/airflow-multicloud/local_run_of_callables.py:29]: was this intentional? same as gpu_cluster_config name

File: examples/dask-preprocessing-and-training/lightgbm_training.py [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+        self.mae = mean_absolute_error(y_test, y_pred)
+        print(f"Mean Absolute Error: {self.mae}")
+
+        self.mse = mean_squared_error(y_test, y_pred)
+        print(f"Mean Squared Error: {self.mse}")
+
+    def return_model_details(self):
+        return {"features": self.features, "mse": self.mse, "mae": self.mae}
+
+    def save_model(self, path, upload_to_s3=False):
```
Comment (jlewitt1) [inline examples/dask-preprocessing-and-training/lightgbm_training.py:88]: nit - maybe `upload_to_storage` or something more generic?

File: examples/dask-preprocessing-and-training/permissions.yaml (hunk 3 above)
Comment (py-rh) [reply examples/dask-preprocessing-and-training/permissions.yaml:1]: permissions in GCP required - don't need it here, but it was in a folder for someone else. I will remove it

File: examples/hpo/hpo.py (hunk 4 above)
Comment (py-rh) [reply examples/hpo/hpo.py:25]: it runs without so IDK

File: examples/dask-preprocessing-and-training/launch_lightgbm_dask_aws.py [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+            }
+        )
+    )
+
+    cluster = rh.cluster(
+        name=cluster_name,
+        instance_type="r5d.xlarge",
+        num_nodes=num_nodes,
+        provider="aws",
+        launcher_type="local",
```
[the remark this answers is not in the document]
Comment (py-rh) [reply examples/dask-preprocessing-and-training/launch_lightgbm_dask_aws.py]: fixed

**Pull request merged**